docs/dev/ci/github_actions/docker_images.md

```diff
@@ -1,5 +1,76 @@
 # Overview of the Docker Images used in the OpenVINO GitHub Actions CI
 
+Most of the workflows in the OpenVINO GHA CI are using [self-hosted machines with dynamically spawned runners](./runners.md) to execute jobs. 
+
+To avoid corruption of the runners and machines, the workflows utilize various Docker images that introduce a layer of protection for the self-hosted machines.
+
+The Docker images are specified for each job using the `container` key. See the [GHA documentation](https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container) for the syntax reference.
+
+An example `Build` job from the `linux.yml` workflow:
+```yaml
+Build:
+  ...
+  container:
+    image: openvinogithubactions.azurecr.io/dockerhub/ubuntu:20.04
+    volumes:
+      - /mount:/mount
+  ...
+```
+
+The `openvinogithubactions.azurecr.io/dockerhub/ubuntu:20.04` Docker image is used for this job.
+
+Additionally, it is possible to make the caches available in the Docker containers using the `volumes` key. 
+Read more about the available caches and how to choose one [here](./caches.md).
+
 ## Available Docker Images
 
+The jobs in the workflows utilize appropriate Docker images based on a job's needs. 
+
+As the self-hosted machines are hosted in [Microsoft Azure](https://azure.microsoft.com/en-us), 
+it is optimal to use the Docker images hosted in an instance of [Azure Container Registry (ACR)](https://azure.microsoft.com/en-us/products/container-registry).
+
+The ACR used for the OpenVINO GHA CI is `openvinogithubactions.azurecr.io`.
+
+Up- and downsides of this approach are:
```

Comment:
[minor] Maybe it would be better to list pros & cons separately? Something like
>pros:
> * ...
> * ...
>
>cons: 
> * ...

It's a bit hard to read "**\+** - ..." and "**\-** -"